src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -163,6 +165,49 @@ public void Startup(StartupParams startupParams)
             noNetworkMode = startupParams.NoNetworkMode;
         }
 
+        /// <summary>
+        /// When a new engine is added from a package, add its dependency to the respective package in the dictionary.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        internal void PythonEngineAdded(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add)
+            {
+                try
+                {
+                    var assem = e.NewItems[0]?.GetType().Assembly;
+                    if (assem == null) return;
+
+                    var assemLoc = assem.Location;
+                    foreach (var pkg in PackageLoader.LocalPackages)
+                    {
+                        if (assemLoc.StartsWith(pkg.RootDirectory))
+                        {
+                            if (NodePackageDictionary.ContainsKey(assem.FullName))
+                            {
+                                var assemName = AssemblyName.GetAssemblyName(assem.Location);
+                                OnMessageLogged(LogMessage.Info(
+                                    string.Format("{0} contains the python engine library {1}, which has already been loaded " +
```

Comment:
nitpick IMO more readable to use string interpolation 
https://stackoverflow.com/questions/33236592/multiline-c-sharp-interpolated-string-literal